test/results_test.exs

```diff
@@ -4,35 +4,41 @@ defmodule Wanda.ResultsTest do
 
   import Wanda.Factory
 
+  alias Wanda.Execution.Target
+
   alias Wanda.Results
 
   alias Wanda.Results.ExecutionResult
 
-  describe "Appending Execution Results to the history log" do
-    test "should correctly add an item to an empty log" do
+  describe "Creating an Execution" do
+    test "should correctly create a running execution in an empty system" do
       execution_id = UUID.uuid4()
       group_id = UUID.uuid4()
 
-      Results.create_execution_result!(
-        build(
-          :result,
-          execution_id: execution_id,
-          group_id: group_id,
-          result: :passing
-        )
-      )
+      [
+        %Target{
+          agent_id: agent_id,
+          checks: checks
+        }
+      ] = targets = build_list(1, :target)
+
+      Results.create_execution_result!(execution_id, group_id, targets)
 
       assert %ExecutionResult{
                execution_id: ^execution_id,
                group_id: ^group_id,
-               payload: %{
-                 "result" => "passing",
-                 "check_results" => [_ | _]
-               }
+               status: :running,
+               payload: %{},
+               targets: [
```

Comment:
I think we could simply `targets: ^targets` here, to not have this explicit pins

Reply:
I am afraid that won't match as we have a list of `Wanda.Execution.Target` on one side and a list of simpler structs `%{}` on the other side.